src/ducks/Alert/components/AlertModalContent/StepsContent/entities/WalletAndConditionsSelector/EventSelector/EventSelector.js

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect, useState } from 'react'
+import cx from 'classnames'
+import { useField } from 'formik'
+import { getAddressInfrastructure } from 'webkit/utils/address'
+import { WALLET_EVENTS } from './constants'
+import styles from './EventSelector.module.scss'
+
+const EventSelector = ({ address, onEventChange, setSelectedAsset }) => {
+  const [, { value: settings }, { setValue: setSettings }] = useField('settings')
+  const hasInitSettings = !!(settings.target && settings.target.address)
+  const initialEvent = WALLET_EVENTS.find((event) => event.settings.type === settings.type)
```

Comment:
Better to do the search inside the `useMemo`, otherwise it will be repeated on every render